main.c

```diff
@@ -1,25 +1,117 @@
 #include "pico/stdlib.h"
 #include "pico/cyw43_arch.h"
+#include "hardware/pwm.h"
+#include "modes.h"
 
 
 #define GPIO_ON 1
 #define GPIO_OFF 0
+#define PWM_GPIO_PIN 14
+#define BUTTON_GPIO_PIN 15
+#define WRAP_POINT_2KHZ 62500
 
 
+    //Set initial vales for variables
+    int mode = 0;
+    int timer_ms = 0;
+    int debounce_timer_ms = 0;
+
+
+    //Irq callback handler
+    void button_callback_handler(uint gpio, uint32_t event){
+
+        //switch mode
+        if(debounce_timer_ms == 0){
+            switch_mode(&mode, &timer_ms);
+        }
+        debounce_timer_ms = 300;
```

Comment:
I think it's a bit too much, let's make it something like 15/30ms [ Check if it works for you empirically

Reply:
Also please change name, it actually isn't milliseconds, lets call it debounce_counter or debounce_timer. It may be misleading in the future.